bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -105,6 +107,65 @@ impl Lambda {
         }
     }
 
+    pub(crate) fn generate_network_enhanced_metrics(
+        network_offset: NetworkData,
+        network_data: NetworkData,
+        aggr: &mut std::sync::MutexGuard<Aggregator>,
+    ) {
+        let rx_bytes = network_data.rx_bytes - network_offset.rx_bytes;
+        let tx_bytes = network_data.tx_bytes - network_offset.tx_bytes;
+        let total_network = rx_bytes + tx_bytes;
+
+        let metric = Metric::new(
+            constants::RX_BYTES_METRIC.into(),
+            MetricValue::distribution(rx_bytes),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("failed to insert rx_bytes metric: {}", e);
+        }
+
+        let metric = Metric::new(
+            constants::TX_BYTES_METRIC.into(),
+            MetricValue::distribution(tx_bytes),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("failed to insert tx_bytes metric: {}", e);
+        }
+
+        let metric = Metric::new(
+            constants::TOTAL_NETWORK_METRIC.into(),
+            MetricValue::distribution(total_network),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("failed to insert total_network metric: {}", e);
+        }
+    }
+
+    pub fn set_network_enhanced_metrics(&self, network_offset: Option<NetworkData>) {
+        if !self.config.enhanced_metrics {
+            return;
+        }
+
+        if let Some(offset) = network_offset {
+            let mut aggr: std::sync::MutexGuard<Aggregator> =
+                self.aggregator.lock().expect("lock poisoned");
+
+            match proc::get_network_data() {
+                Ok(data) => {
+                    Self::generate_network_enhanced_metrics(offset, data, &mut aggr);
+                }
+                Err(_e) => {
+                    debug!("Could not emit network enhanced metrics");
+                }
+            }
+        } else {
+            debug!("Could not emit network enhanced metrics");
```

Comment:
Maybe we change the wording? This is not accurate, it's more like no data, so we didn't send anything

Reply:
True, would "Could not generate network enhanced metrics" or "Could not find data to generate network enhanced metrics" sound better

Reply:
The second looks better!